config/base/src/lib.rs

```diff
@@ -478,7 +480,55 @@ pub mod proxy {
         ///
         /// # Errors
         /// - Fails if the deserialization of any field fails.
-        fn from_env() -> Self::ReturnValue;
+        fn from_env<F: FetchEnv>(fetcher: &F) -> Self::ReturnValue;
+
+        /// Implementation of [`Self::from_env`] using [`std::env::var`].
+        fn from_std_env() -> Self::ReturnValue {
+            struct FetchStdEnv;
+
+            impl FetchEnv for FetchStdEnv {
+                fn fetch<K: AsRef<std::ffi::OsStr>>(
+                    &self,
+                    key: K,
+                ) -> Result<String, std::env::VarError> {
+                    std::env::var(key)
+                }
+            }
+
+            Self::from_env(&FetchStdEnv)
+        }
+    }
+
+    /// Errors that might occur during the loading data from environment
+    /// using [`LoadFromEnv`] trait.
+    #[derive(Debug, Error)]
+    pub enum LoadFromEnvError {
+        /// JSON5 deserialization error
+        #[error("Failed to deserialize env var `{}` as JSON5: {}", .env_var, .error)]
+        Json5 {
+            /// The environment variable name
+            env_var: &'static str,
+            /// The actual parsing error
+            error: json5::Error,
+        },
+    }
```

Comment:
I would like to re-use `iroha_config_base::derive::Error::Json5`:

```rust
        /// Used in [`LoadFromDisk`](`crate::proxy::LoadFromDisk`) trait for deserialization errors
        #[error("Deserializing JSON failed")]
        #[serde(skip)]
        Json5(#[from] json5::Error),
```

But it doesn't have info about the environment variable.

Reply:
I agree. Unfortunately, we don't have a universal mechanism to specify field locations in errors we throw:

- #3470 

So it makes more sense to reuse existing `FieldDeserialization` and not bother about `env_var` for now. It should be fixed later all-at-once with #3470.

Reply:
Fixed